src/map/battle.cpp

```diff
@@ -393,7 +393,16 @@ int battle_delay_damage(t_tick tick, int amotion, struct block_list *src, struct
 		damage = 0;
 	}
 
-	if ( !battle_config.delay_battle_damage || amotion <= 1 ) {
+	// Check for delay battle damage config
+	// If the synchronize damage feature is activated then disabling the config should only affect skills (for monsters)
+	if (!battle_config.delay_battle_damage && (!battle_config.synchronize_damage || skill_id != 0 || src->type != BL_MOB))
+		amotion = 1;
+	// If clientamotion is 1 or lower for monsters, we can apply damage directly for normal attacks and don't need to create a timer
+	else if (battle_config.synchronize_damage && skill_id == 0 && src->type == BL_MOB && status_get_clientamotion(src) <= 1)
+		amotion = 1;
+
+	// Skip creation of timer
+	if (amotion <= 1) {
```

Comment:
```suggestion
	// The client refuses to display animations slower than 1x speed
	// So we need to shorten AttackMotion to be in-sync with the client in this case
	if( battle_config.synchronize_damage && skill_id == 0 && src->type == BL_MOB && amotion > status_get_clientamotion(src) ){
		amotion = status_get_clientamotion(src);
	}

	if ( !battle_config.delay_battle_damage || amotion <= 1 ) {
```

Reply:
I don't understand this change, doesn't this just remove the whole feature of making sychronize_damage work with delay_battle_damage?

Reply:
Either you want to delay or you want to synchronize. You cannot have both.
If whoever enables your new setting, they want synchronized damage and not delayed damage.
So instead of hiding that we give the new feature priority, we just straight forward go for it.

(And additionally we dont touch the old feature - we may have to find out, when it was broken - or if it has always been for any damage, but outside of the scope of this PR)

Reply:
Hmm, but it's important to give the player the option to combine the two settings. Basically battle_sychronize_damage only affects normal skills so of course if you enable this you want normal attacks to be synchronized, which should take priority over the other feature (I'll make it clearer in the description that it takes priority).

At the same time, we want to give the user the choice if they want to delay skill damage (and player damage) or not.

The current implementation from me does not affect the other feature as long as you keep it disabled, so it isn't a problem of backward compatibility.